scripts/main/album.js

```diff
@@ -699,326 +768,308 @@ album.setSorting = function (albumID) {
 	});
 };
 
-album.setPublic = function (albumID, e) {
-	let password = "";
-
-	if (!basicModal.visible()) {
-		let msg = lychee.html`
-			<form>
-				<div class='switch'>
-					<label>
-						${lychee.locale["ALBUM_PUBLIC"]}:&nbsp;
-						<input type='checkbox' name='is_public'>
-						<span class='slider round'></span>
-					</label>
-					<p>${lychee.locale["ALBUM_PUBLIC_EXPL"]}</p>
-				</div>
-				<div class='choice'>
-					<label>
-						<input type='checkbox' name='grants_full_photo'>
-						<span class='checkbox'>${build.iconic("check")}</span>
-						<span class='label'>${lychee.locale["ALBUM_FULL"]}</span>
-					</label>
-					<p>${lychee.locale["ALBUM_FULL_EXPL"]}</p>
-				</div>
-				<div class='choice'>
-					<label>
-						<input type='checkbox' name='requires_link'>
-						<span class='checkbox'>${build.iconic("check")}</span>
-						<span class='label'>${lychee.locale["ALBUM_HIDDEN"]}</span>
-					</label>
-					<p>${lychee.locale["ALBUM_HIDDEN_EXPL"]}</p>
-				</div>
-				<div class='choice'>
-					<label>
-						<input type='checkbox' name='is_downloadable'>
-						<span class='checkbox'>${build.iconic("check")}</span>
-						<span class='label'>${lychee.locale["ALBUM_DOWNLOADABLE"]}</span>
-					</label>
-					<p>${lychee.locale["ALBUM_DOWNLOADABLE_EXPL"]}</p>
-				</div>
-				<div class='choice'>
-					<label>
-						<input type='checkbox' name='is_share_button_visible'>
-						<span class='checkbox'>${build.iconic("check")}</span>
-						<span class='label'>${lychee.locale["ALBUM_SHARE_BUTTON_VISIBLE"]}</span>
-					</label>
-					<p>${lychee.locale["ALBUM_SHARE_BUTTON_VISIBLE_EXPL"]}</p>
-				</div>
-				<div class='choice'>
-					<label>
-						<input type='checkbox' name='has_password'>
-						<span class='checkbox'>${build.iconic("check")}</span>
-						<span class='label'>${lychee.locale["ALBUM_PASSWORD_PROT"]}</span>
-					</label>
-					<p>${lychee.locale["ALBUM_PASSWORD_PROT_EXPL"]}</p>
-					<input class='text' name='passwordtext' type='text' placeholder='${lychee.locale["PASSWORD"]}' value=''>
-				</div>
-				<div class='hr'><hr></div>
-				<div class='switch'>
-					<label>
-						${lychee.locale["ALBUM_NSFW"]}:&nbsp;
-						<input type='checkbox' name='is_nsfw'>
-						<span class='slider round'></span>
-					</label>
-					<p>${lychee.locale["ALBUM_NSFW_EXPL"]}</p>
-				</div>
-			</form>
-		`;
+/**
+ * Sets the accessibility attributes of an album.
+ *
+ * **ATTENTION**: Despite its name this method does not only set the
+ * `is_public` attribute of the album, but configures most of the attributes
+ * which contribute to the accessibility of an album.
+ *
+ * TODO: Find a better name for the method.
```

Comment:
setAccessPolicy ?

Reply:
Great idea, I'll take it.

Reply:
I renamed it to `setProtectionPoilcy`. After second thoughts, I did not like `setAccessPolicy`, because we do not only set the accessibility but also visibility. Keep in mind, that the term accessibility is already taken.

As in Harry Potter: Platform 9 3/4 is accessible, but invisible. You must run straight to the wall between 9 and 10. (=You need the direct link).

And both accessibility and viability form browsability. So I needed another term for the dialog title and API call which works as an umbrella term and I came up with "protection policy".